controller/controller_server/csi_controller_server.py

```diff
@@ -90,6 +91,11 @@ def CreateVolume(self, request, context):
                     required_bytes = min_size
                     logger.debug("requested size is 0 so the default size will be used : {0} ".format(
                         required_bytes))
+
+                if system_id and system_id != array_mediator.identifier:
+                    raise ValidationException(
+                        controller_messages.system_id_mismatch_message.format(system_id, array_mediator.identifier))
```

Comment:
shouldn't it be the first thing to check in the `with`?

Reply:
is it a stronger problem than the size error (the part before)?

Reply:
good question.
in a highly unlikely scenario, if a user has different storage types (xiv/svc/ds8k) and got the management address wrong, the size limitations may not be relevant

Reply:
fair enough, done